src/parser/ModuleParser.mts

```diff
@@ -20,7 +20,22 @@ export abstract class ModuleParser extends StatementParser {
     if (this.test(Token.STRING)) {
       node.ModuleSpecifier = this.parsePrimaryExpression();
     } else {
-      node.ImportClause = this.parseImportClause();
+      const importClause = this.parseImportClause();
+      if (
+        this.feature('import-defer')
+        && importClause.ImportedDefaultBinding?.sourceText() === 'defer'
```

Comment:
I'm not sure I understanded this part, does it means it parses like `import defer ` as part of `import defer from ...` (import default) and repurpose it as import phase? If so it looks hacky, it will be better if you can `this.peek()` to branch the parser.

Reply:
Yeah I can do `.peek()`, I was used to what we do in Babel's parser where we try to avoid peeking for performance reasons.